src/__tests__/fake-timers.js

```diff
@@ -1,20 +1,26 @@
 import {waitFor, waitForElementToBeRemoved} from '..'
 import {render} from './helpers/test-utils'
 
-async function runWaitFor({time = 300} = {}, options) {
+async function runWaitFor({time = 300, advanceTimers = true} = {}, options) {
   const response = 'data'
   const doAsyncThing = () =>
     new Promise(r => setTimeout(() => r(response), time))
   let result
   doAsyncThing().then(r => (result = r))
-
+  if (advanceTimers) {
+    jest.advanceTimersByTime(time)
```

Comment:
Shouldn't `waitFor` take care of this? 

@timdeschryver I think this is fixed by https://github.com/testing-library/dom-testing-library/pull/966 and https://github.com/testing-library/dom-testing-library/pull/962 without needing to change these particular tests. Can we close this PR in favor of https://github.com/testing-library/dom-testing-library/pull/962?

Reply:
Yes of course.
Thanks for getting back to this PR, because I forgot about it.

Reply:
> Shouldn't waitFor take care of this?

It could've been a bug in `waitFor` then, the reason why I added this was because the timers weren't advancing on their own. So, I thought that I had to do that manually.